tests/test_scalars.py

```diff
@@ -24,17 +39,685 @@
+    # pyright can't know that as_dt_timedelta - td calls td.__rsub__(as_dt_timedelta),
+    # not as_dt_timedelta.__sub__
+    # https://github.com/microsoft/pyright/issues/4088
+    check(
+        assert_type(
+            as_dt_timedelta - td,  # pyright: ignore[reportGeneralTypeIssues]
+            pd.Timedelta,
+        ),
+        pd.Timedelta,
+    )
+    check(assert_type(as_timedelta64 - td, pd.Timedelta), pd.Timedelta)
+    check(assert_type(as_timedelta_index - td, pd.TimedeltaIndex), pd.TimedeltaIndex)
+    check(
+        assert_type(as_timedelta_series - td, TimedeltaSeries), pd.Series, pd.Timedelta
+    )
+    check(assert_type(as_period_index - td, pd.PeriodIndex), pd.PeriodIndex)
+    check(assert_type(as_datetime_index - td, pd.DatetimeIndex), pd.DatetimeIndex)
+    check(
+        assert_type(
+            as_ndarray_td64 - td,
+            npt.NDArray[np.timedelta64],
+        ),
+        np.ndarray,
+        np.timedelta64,
+    )
+    check(
+        assert_type(
+            as_ndarray_dt64 - td,
+            npt.NDArray[np.datetime64],
+        ),
+        np.ndarray,
+        np.datetime64,
+    )
+    check(assert_type(as_nat - td, NaTType), NaTType)
+
+
+def test_timedelta_mul_div() -> None:
+    td = pd.Timedelta("1 day")
+
+    with pytest.warns(FutureWarning):
+        i_idx = cast(pd.Int64Index, pd.Index([1, 2, 3], dtype=int))
+        f_idx = cast(pd.Float64Index, pd.Index([1.2, 2.2, 3.4], dtype=float))
+
+    np_intp_arr: npt.NDArray[np.integer] = np.array([1, 2, 3])
+    np_float_arr: npt.NDArray[np.floating] = np.array([1.2, 2.2, 3.4])
+
+    md_int = 3
+    md_float = 3.5
+    md_ndarray_intp = np_intp_arr
+    md_ndarray_float = np_float_arr
+    mp_series_int = pd.Series([1, 2, 3], dtype=int)
+    md_series_float = pd.Series([1.2, 2.2, 3.4], dtype=float)
+    md_int64_index = i_idx
+    md_float_index = f_idx
+    md_timedelta_series = pd.Series(pd.timedelta_range("1 day", periods=3))
+
+    check(assert_type(td * md_int, pd.Timedelta), pd.Timedelta)
+    check(assert_type(td * md_float, pd.Timedelta), pd.Timedelta)
+    check(
+        assert_type(td * md_ndarray_intp, npt.NDArray[np.timedelta64]),
+        np.ndarray,
+        np.timedelta64,
+    )
+    check(
+        assert_type(td * md_ndarray_float, npt.NDArray[np.timedelta64]),
+        np.ndarray,
+        np.timedelta64,
+    )
+    check(assert_type(td * mp_series_int, TimedeltaSeries), pd.Series, pd.Timedelta)
+    check(assert_type(td * md_series_float, TimedeltaSeries), pd.Series, pd.Timedelta)
+    check(assert_type(td * md_int64_index, pd.TimedeltaIndex), pd.TimedeltaIndex)
+    check(assert_type(td * md_float_index, pd.TimedeltaIndex), pd.TimedeltaIndex)
+
+    check(assert_type(md_int * td, pd.Timedelta), pd.Timedelta)
+    check(assert_type(md_float * td, pd.Timedelta), pd.Timedelta)
+    check(assert_type(md_ndarray_intp * td, np.ndarray), np.ndarray)
+    check(assert_type(md_ndarray_float * td, np.ndarray), np.ndarray)
```

Comment:
For the above two tests, add `np.timedelta64` to the `check` arguments

Reply:
Fixed